cmd/compile/main.go

```diff
@@ -65,24 +87,34 @@ func run() error {
 	if err := dockerRun("make", "bcmrpi_defconfig"); err != nil {
 		return err
 	}
+	// disable all modules (mod2noconfig is not in RPi 5.15)
+	if err := dockerRun("sed", "s/=m$/=n/i", "-i", ".config"); err != nil {
+		return err
+	}
 	if err := chown(".config"); err != nil {
 		return err
 	}
 
-	// adjust config to add CONFIG_SQUASHFS and CONFIG_IPV6
+	// write additions to config that we need
 	configPath := filepath.Join(kernelFolder, ".config")
-	err = adjustTextFile(configPath, func(line string) bool {
-		return strings.HasPrefix(line, "CONFIG_SQUASHFS=") || strings.HasPrefix(line, "CONFIG_IPV6=")
-	}, []string{
-		"CONFIG_SQUASHFS=y",
-		"CONFIG_IPV6=y",
-	})
+	f, err := os.OpenFile(configPath, os.O_APPEND|os.O_WRONLY, 0644)
 	if err != nil {
 		return err
 	}
+	defer f.Close()
+	if _, err := f.Write([]byte(configAddendum)); err != nil {
+		return err
+	}
+	if err := f.Close(); err != nil {
+		return err
+	}
+	// fix the config with our addendums
+	if err := dockerRun("make", "olddefconfig"); err != nil {
+		return err
+	}
```

Comment:
I think it would be better to adjust the `adjustTextFile` function instead of:
- running `sed`
- appending some lines
- running `make olddefconfig` (which prints out some warnings)


What do you think of a `adjustConfig` function which performs all 3 operations in one go:
```
// adjustConfig will disable all settings set as module and adjust the given config
func adjustConfig(path string, config map[string]string) {
	// read whole file
	for _, line := range strings.Split(string(b), "\n") {
		key, value, _ := strings.Cut(line, "=")

		if _, ok := config[key]; ok { // custom setting will be appended
			continue
		}
		if value == "m" { // modules should be disabled
			line = key + "=n"
		}

		_, err = w.WriteString(line + "\n")
		if err != nil {
			return err
		}
	}
	for key, value := range config {
		_, err = w.WriteString(key + "=" + value + "\n")
		if err != nil {
			return err
		}
	}
	// flush and close
}
```

Reply:
I am much more comfortable running `olddefconfig` for two reasons:
1. If you enable any options that have dependencies, this will do that correctly and accurately for you, as opposed to random sed magic and then just running the build hoping for the best (a working kernel).
2. This is exactly what the vanilla kernel does, so it seems to be the proven solution.

Reply:
As for the warnings, building the Linux kernel I've kind of gotten used to warnings in the config generation phase, they seem hard to avoid even when you interactively configure the kernel sometimes.
The only way to avoid them and get the comfort of point `1` above would be to swap back to the `adjustFile` approach and then still run `olddefconfig`, which might still yield some warnings (which, to me, is pretty much equivalent to `sed`, except in golang rather than command line)